### DISCUSSION TITLE
Delete remote branch on delete local?

### DISCUSSION THREAD

**Discussion by scascalespAV** [gitextensions/gitextensions#10510] [Q&A]
Hello
Exist or is possible to add some check in options to delete remote branch when i delete one local branch?
Thanks

**Comment by pmiossec** [comment]
No, this option doesn't exist in GitExtensions.

I have a dirty and basic hack in my custom build to do the opposite (delete a local tracking branch when deleting the remote branch but that ends up doing the same thing 😉 ) and it is very convenient...

But doing it well to upstream it with all the error management and a good UX is a time consuming task that I never had the motivation to do 😢 

What it looks like at the moment (fyi, will surely be broken in a near future when I will rebase my custom branch) : https://github.com/gitextensions/gitextensions/commit/35f723f7d7fb3d0cd57770337d0cbc25c87e448c

**Comment by RussKie** [comment]
Deletion of a remote branch is a destructive operation, so we won't be
providing functionality to make it easy.

If it is something you **really** need you can create a script to that
effect. See under Settings > Scripts.

**Reply by pmiossec** [reply]
> Deletion of a remote branch is a destructive operation, so we won't be providing functionality to make it easy.

In fact, in GE, we can delete a remote branch as any other branch.
There is just a checkbox to ensure that the user really understand that he will delete the branch in the remote repository.

![image](https://user-images.githubusercontent.com/460196/207267546-3de6c057-21b8-4b2c-954a-3956f1c8501b.png)

Si, I agree with you that it's dangerous and that we **must** not allow a user to delete a remote while deleting a local branch.
But IMO, it could make sense to let the user delete the local tracking branch **when he decides to delete the remote branch** (to clean a PR merged)

Like I did in my custom version (where I just open the delete local branch popup with the branch already selected)
![image](https://user-images.githubusercontent.com/460196/207268278-4cea89b8-a7ad-480e-83b6-8f6e7c42231d.png)

**Reply by scascalespAV** [reply]
Hello. Where i can find this option? I have this:
![image](https://user-images.githubusercontent.com/110404394/207289734-cb2008e3-4cf3-4d23-b4a8-d9aa756104b7.png)

**Reply by pmiossec** [reply]
@scascalespAV I just wrote 2 times that it is in my custom build  😉

**Reply by RussKie** [reply]
> But IMO, it could make sense to let the user delete the local tracking branch **when he decides to delete the remote branch** (to clean a PR merged)

👍

**Reply by scascalespAV** [reply]
@pmiossec  where i can download it?

**Reply by pmiossec** [reply]
nowhere. it's a custom build. But nothing prevent you to build it yourself...
